src/VPolygon.jl

```diff
@@ -276,6 +277,123 @@ function ∈(x::AbstractVector{N}, P::VPolygon{N})::Bool where {N<:Real}
     return true
 end
 
+"""
+    rand(::Type{VPolygon}; [N]::Type{<:Real}=Float64, [dim]::Int=2,
+         [rng]::AbstractRNG=GLOBAL_RNG, [seed]::Union{Int, Nothing}=nothing
+        )::VPolygon{N}
+
+Create a random polygon in vertex representation.
+
+### Input
+
+- `VPolygon`     -- type for dispatch
+- `N`            -- (optional, default: `Float64`) numeric type
+- `dim`          -- (optional, default: 2) dimension
+- `rng`          -- (optional, default: `GLOBAL_RNG`) random number generator
+- `seed`         -- (optional, default: `nothing`) seed for reseeding
+- `num_vertices` -- (optional, default: `-1`) number of vertices of the
+                    polygon (see comment below)
+
+### Output
+
+A random polygon in vertex representation.
+
+### Algorithm
+
+We follow the idea [here](https://stackoverflow.com/a/47358689) based on
+[P. Valtr. Probability that n random points are in convex
+position](https://link.springer.com/article/10.1007%2FBF01271274).
+
+The number of vertices can be controlled with the argument `num_vertices`.
+For a negative value we choose a random number in the range `1:10`.
+"""
+function rand(::Type{VPolygon};
+              N::Type{<:Real}=Float64,
+              dim::Int=2,
+              rng::AbstractRNG=GLOBAL_RNG,
+              seed::Union{Int, Nothing}=nothing,
+              num_vertices::Int=-1
+             )::VPolygon{N}
+    @assert dim == 2 "a polygon must have dimension 2"
+    rng = reseed(rng, seed)
+    if num_vertices < 0
+        num_vertices = rand(1:10)
+    end
+
+    # special cases, 0 or 1 vertex
```

Comment:
this corner case can be treated in functions that rely on this function, eg. 

```julia
julia> rand(HPolygon, num_constraints=2).constraints
4-element Array{HalfSpace{Float64},1}:
 HalfSpace{Float64}([0.398179, 0.371021], -0.5601983753144615)
 HalfSpace{Float64}([-0.371021, 0.398179], -1.5636591141726837)
 HalfSpace{Float64}([-0.398179, -0.371021], 0.8564018279869186)
 HalfSpace{Float64}([0.371021, -0.398179], 1.5636591141726837)
```

should instead raise an error

Reply:
I do not understand exactly, but I now raise an error if the number of constraints is two or less.